[NFC][StructurizeCFG] Add a test that can crash StructurizeCFG pass

shiltian opened the pull request:
I tried to fix it in #124051 but failed to do so. This PR adds the test and
marks it as xfail.

shiltian commented:
* **#126087** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/126087?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/126087?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

arsenm reviewed (commented):
Can you also open a tracking issue and add a reference to the test (like in the file name)

shiltian committed:
[NFC][StructurizeCFG] Add a test that can crash StructurizeCFG pass

I tried to fix it in #124051 but failed to do so. This PR adds the test and
marks it as xfail.

shiltian commented:
> Can you also open a tracking issue and add a reference to the test (like in the file name)

Done.

Prabhuk commented:
This test fails in our (Fuchsia) toolchain CI linux host builders. I am currently investigating this.

```
XPASS: LLVM :: Transforms/StructurizeCFG/simple-structurizecfg-crash.ll (50157 of 58077)
******************** TEST 'LLVM :: Transforms/StructurizeCFG/simple-structurizecfg-crash.ll' FAILED ********************
Exit Code: 0

Command Output (stdout):
--
; ModuleID = '/b/s/w/ir/x/w/llvm-llvm-project/llvm/test/Transforms/StructurizeCFG/simple-structurizecfg-crash.ll'
source_filename = "/b/s/w/ir/x/w/llvm-llvm-project/llvm/test/Transforms/StructurizeCFG/simple-structurizecfg-crash.ll"

define void @foo() {
entry:
  br label %cond.false

cond.true:                                        ; preds = %cond.false
  br label %cond.end

cond.false:                                       ; preds = %entry
  br i1 true, label %cond.true, label %cond.end

cond.end:                                         ; preds = %cond.true, %cond.false
  ret void
}

--
Command Output (stderr):
--
RUN: at line 1: /b/s/w/ir/x/w/llvm_build/bin/opt -S -passes=structurizecfg /b/s/w/ir/x/w/llvm-llvm-project/llvm/test/Transforms/StructurizeCFG/simple-structurizecfg-crash.ll -o -
+ /b/s/w/ir/x/w/llvm_build/bin/opt -S -passes=structurizecfg /b/s/w/ir/x/w/llvm-llvm-project/llvm/test/Transforms/StructurizeCFG/simple-structurizecfg-crash.ll -o -
```

Link to the logs: https://logs.chromium.org/logs/fuchsia/buildbucket/cr-buildbucket/8723305309974990577/+/u/clang/tests/stdout
Link to the bot: https://luci-milo.appspot.com/ui/p/fuchsia/builders/toolchain.ci/clang-host-linux-x64/b8723305798791238737/overview

dyung commented:
I'm also seeing the test XPASS on my bot which builds without assertions enabled: 

https://lab.llvm.org/staging/#/builders/202/builds/1029

arsenm commented on llvm/test/Transforms/StructurizeCFG/simple-structurizecfg-crash.ll:
```diff
@@ -0,0 +1,19 @@
+; RUN: opt -S -passes=structurizecfg %s -o -
+; XFAIL: *
+
```
Requires asserts

shiltian commented:
@dyung 15412d9d83e23ce7de23dc7252391ee6403384af should hopefully fix this.
